ext/opcache/jit/zend_jit_arm64.dasc

```diff
@@ -390,9 +395,9 @@ static bool logical_immediate_p(uint64_t value, uint32_t reg_size)
 |.macro CMP_64_WITH_CONST, reg, val, tmp_reg
 ||	if (val == 0) {
 |		cmp reg, xzr
-||	} else if (((int64_t)(val)) > 0 && ((int64_t)(val)) <= CMP_IMM) {
+||	} else if (arm64_may_encode_imm12((int64_t)(val))) {
 |		cmp reg, #val
-||	} else if (((int64_t)(val)) < 0 && ((int64_t)(val)) >= -CMP_IMM) {
+||	} else if (arm64_may_encode_imm12((int64_t)(-val))) {
```

Comment:
Did you test this?
In any case, it would be great to add test(s) for edge cases.

Reply:
I have run ~4k .phpt test cases under "Zend/tests/, tests/ ext/opcache/tests/jit/", and didn't find any failure.

Good suggestion! I will add several new test cases soon. Thanks.

Reply:
Two test cases are added, and in my local testing, immediate values in legitimate ranges can be encoded as the `imm` field as expected.
1) for `cmp` and `cmn` instructions, I only tested  `CMP_64_WITH_CONST`. Note: I tested that negative values can be encoded into the `imm` field fo `cmn` instruction.
2) for ADD and SUB, I only tested `adds` and `subs` with macro `ADD_SUB_64_WITH_CONST`.

I currently failed to construct test cases to use `CMP_32_WITH_CONST, CMP_64_WITH_CONST_32, ADD_SUB_32_WITH_CONST, ADD_SUB_64_WITH_CONST_32`, but I think `arm64_may_encode_imm12` would work for them as well since these 4 macros share the same encoding logic of `imm12` with `CMP_64_WITH_CONST` and `ADD_SUB_64_WITH_CONST`.